### PR TITLE
dev-python/wxpython: apply upstream fix

### PR DESCRIPTION
**Pull request by t0b3**
Correct types of arguments to ScrolledWindow.SetScrollbars
Related: https://github.com/wxWidgets/Phoenix/pull/2005

### CODE REVIEW

**Comment by t0b3** [comment]
@alexxy ping?

File: dev-python/wxpython/wxpython-4.0.7-r2.ebuild [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+#RESTRICT="!test? ( test )"
+
+# wxPython doesn't seem to be able to optionally disable features. webkit is
+# optionally patched out because it's so huge, but other elements are not,
+# which makes us have to require all features from wxGTK
+RDEPEND="
+	>=x11-libs/wxGTK-3.0.4-r301:${WX_GTK_VER}=[gstreamer,libnotify,opengl,sdl,tiff,webkit?,X]
+	media-libs/libpng:0=
+	media-libs/tiff:0
+	virtual/jpeg:0"
```
Comment (AndrewAmmerlaan) [inline dev-python/wxpython/wxpython-4.0.7-r2.ebuild:34]: Could you also update this deprecated dependency please?

File: dev-python/wxpython/wxpython-4.0.7-r2.ebuild [hunk 2]
```diff
@@ -0,0 +1,92 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+PYTHON_COMPAT=( python3_{8..10} )
```
[the remark this answers is not in the document]
Comment (t0b3) [reply dev-python/wxpython/wxpython-4.0.7-r2.ebuild]: done

File: dev-python/wxpython/wxpython-4.0.7-r2.ebuild [hunk 3]
```diff
@@ -0,0 +1,92 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
```
[the remark this answers is not in the document]
Comment (t0b3) [reply dev-python/wxpython/wxpython-4.0.7-r2.ebuild]: done

File: dev-python/wxpython/wxpython-4.0.7-r2.ebuild (hunk 1 above)
Comment (t0b3) [reply dev-python/wxpython/wxpython-4.0.7-r2.ebuild]: done

File: dev-python/wxpython/wxpython-4.0.7-r2.ebuild [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+		--jobs=$(makeopts_jobs) \
+		--release || die
+}
+
+python_install() {
+	distutils-r1_python_install --skip-build
+}
+
+python_test() {
+	virtx pytest -vv unittests
```
[the remark this answers is not in the document]
Comment (t0b3) [reply dev-python/wxpython/wxpython-4.0.7-r2.ebuild]: done

**Comment by t0b3** [comment]
@AndrewAmmerlaan what would you do about this?

`4.0.7-r2 | SourcingError | failed sourcing ebuild: Invalid implementation in PYTHON_COMPAT: python3_11, (python-utils-r1.eclass, line 146:  called die)`

**Comment by AndrewAmmerlaan** [comment]
> @AndrewAmmerlaan what would you do about this?
> 
> `4.0.7-r2 | SourcingError | failed sourcing ebuild: Invalid implementation in PYTHON_COMPAT: python3_11, (python-utils-r1.eclass, line 146: called die)`

Rebase your PR on the latest from master

**Comment by t0b3** [comment]
done

File: dev-python/wxpython/wxpython-4.0.7-r2.ebuild [hunk 5]
```diff
@@ -0,0 +1,92 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+PYTHON_COMPAT=( python3_{8..11} )
+WX_GTK_VER="3.0-gtk3"
+
+DISTUTILS_IN_SOURCE_BUILD="1"
```
Comment (AndrewAmmerlaan) [inline dev-python/wxpython/wxpython-4.0.7-r2.ebuild:8]: Perhaps we can also try the new PEP517 mode, i.e. `DISTUTILS_USE_PEP517=setuptools`

Comment (t0b3) [reply dev-python/wxpython/wxpython-4.0.7-r2.ebuild]: let's try...

File: dev-python/wxpython/wxpython-4.0.7-r2.ebuild (hunk 2 above)
[the remark this answers is not in the document]
Comment (t0b3) [reply dev-python/wxpython/wxpython-4.0.7-r2.ebuild:5]: hmm, thus I removed 3.11 for now

**Comment by AndrewAmmerlaan** [comment]
Thanks :+1:

**Pull request closed without merging**